src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
+const FULL_HEALTH = 100;
+const BITE_DAMAGE = 50;
+
 class Animal {
-  // write your code here
+  i;
+  static alive = [];
+
+  constructor(name, health = FULL_HEALTH) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  die() {
```

Comment:
[CODE PERFORMANCE] - The `die` method should not be responsible for filtering the `Animal.alive` array. Instead, consider moving the logic to remove the animal from the `Animal.alive` array directly into the `bite` method after checking if the animal's health is less than or equal to 0.